flink-runtime/src/test/java/org/apache/flink/runtime/minicluster/TestingMiniCluster.java

```diff
@@ -66,6 +67,17 @@ public TestingMiniCluster(TestingMiniClusterConfiguration miniClusterConfigurati
 		return super.getDispatcherResourceManagerComponents();
 	}
 
+	@Nonnull
+	@Override
+	public CompletableFuture<Void> terminateTaskExecutor(int index) {
+		return super.terminateTaskExecutor(index);
+	}
+
+	@Override
+	public void startTaskExecutor(boolean localCommunication) throws Exception {
```

Comment:
Doesn't look like user friendly API. Shouldn't `MiniCluster` set this flag depending on the configuration? Wouldn't it be enough to expose a signature such as `public void startTaskExecutor()`

Reply:
Well, `MiniCluster` cannot know whether to use local communication or not at start time, because `TestingMiniCluster` allows to start new `TaskExecutors`. Thus, the only option would be to always set `localCommunication` to false in the case of the `MiniCluster`.

Reply:
I introduced a new method `useLocalCommunication` which can be overriden by the `TestingMiniCluster` to always set local communication to `false`.